### ISSUE TITLE
[benchmarks] Resource-Bound Microbenchmarks

### ISSUE THREAD

**Issue by lemosep** [ppenna/nanvix#310]
Right now there's no resource-bound applications in the code base that grants resource performance analysis. 

A possible solution is to include a `test` folder containing different rust applications that are passed as a flag on the [Makefile](https://github.com/nanvix/kernel/blob/f3dccf68cd62cc50a8eedae156613292bac3bf3f/Makefile) to run the kernel. Conditional compilation may also be required so that the kernel knows it'll be executing something.

**Comment by ppenna** [comment]
@lemosep moving this issue to root repository, as applications should not be introduced in kernel layer.

**Comment by lemosep** [comment]
> @lemosep moving this issue to root repository, as applications should not be introduced in kernel layer.

Indeed, it will run on userspace, although the majority of changes will affect the kernel codebase e.g. create and run the process. Wouldn't that need to be assured by the kernel?

**Comment by ppenna** [comment]
> although the majority of changes will affect the kernel codebase e.g. create and run the process.

A user application (benchmark) should not require changes nor affect the kernel.

As a side remark: you should consider listing down the tasks in this issue. That will help you to narrow down the involved steps.

**Comment by lemosep** [comment]
@ppenna Maybe dividing this issue in three different ones, each for one type of benchmark. This way, it'll be easier to discuss, WDYT?
